pkg/controller/jobframework/setup.go

```diff
@@ -21,16 +21,26 @@ import (
 	"errors"
 	"fmt"
 	"os"
+	"time"
 
 	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/util/workqueue"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
 
 	"sigs.k8s.io/kueue/pkg/controller/jobs/noop"
 )
 
+type fetchRestMapperFunc func(mgr ctrl.Manager, gvk schema.GroupVersionKind) (bool, error)
```

Comment:
```suggestion
type fetchRestMapperFunc func(mgr ctrl.Manager, gvk schema.GroupVersionKind) error
```

It doesn't look like the boolean will be true when there is an error, so no need for 2 outputs.

Reply:
This is true, but there is a scenario where it can return false with no errors i.e., if it is an `IsNoMatchError`, it is fine and expected to have missing job frameworks. It could also return false with errors if it's a different error such as issues with fetching the RESTMapper itself. I've modified the function variable to reflect this explicitly.
```
	var defaultCheckAPIAvailable = func(mgr ctrl.Manager, gvk schema.GroupVersionKind) (bool, error) {
		err := restMappingExists(mgr, gvk)
		if meta.IsNoMatchError(err) {
			return false, nil
		}
		if err != nil {
			return false, err
		}
		return true, nil
	}